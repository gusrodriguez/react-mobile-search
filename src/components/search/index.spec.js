
import { shallow } from 'enzyme';
import React from 'react';
import Search from './';

describe('Search', () => {
  let wrapper;
  let props;
  it('should match snapshot', () => {
    props = {
        search: jest.fn(),
      };
    wrapper = shallow(<Search search={props.search} />); 
    expect(wrapper).toMatchSnapshot();
  });
});
