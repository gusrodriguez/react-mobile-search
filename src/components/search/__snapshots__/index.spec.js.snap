// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    search={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        className="search_autosuggest"
        role="search"
      >
        <Autosuggest
          alwaysRenderSuggestions={false}
          focusInputOnSuggestionClick={true}
          getSuggestionValue={[Function]}
          highlightFirstSuggestion={false}
          id="1"
          inputProps={
            Object {
              "onChange": [Function],
              "placeholder": "What would you like to do?",
              "value": "",
            }
          }
          multiSection={false}
          onSuggestionsClearRequested={[Function]}
          onSuggestionsFetchRequested={[Function]}
          renderInputComponent={[Function]}
          renderSuggestion={[Function]}
          renderSuggestionsContainer={[Function]}
          shouldRenderSuggestions={[Function]}
          suggestions={Array []}
          theme={
            Object {
              "container": "react-autosuggest__container",
              "containerOpen": "react-autosuggest__container--open",
              "input": "react-autosuggest__input",
              "inputFocused": "react-autosuggest__input--focused",
              "inputOpen": "react-autosuggest__input--open",
              "sectionContainer": "react-autosuggest__section-container",
              "sectionContainerFirst": "react-autosuggest__section-container--first",
              "sectionTitle": "react-autosuggest__section-title",
              "suggestion": "react-autosuggest__suggestion",
              "suggestionFirst": "react-autosuggest__suggestion--first",
              "suggestionHighlighted": "react-autosuggest__suggestion--highlighted",
              "suggestionsContainer": "react-autosuggest__suggestions-container",
              "suggestionsContainerOpen": "react-autosuggest__suggestions-container--open",
              "suggestionsList": "react-autosuggest__suggestions-list",
            }
          }
        />
      </form>,
      "className": "search",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Autosuggest
          alwaysRenderSuggestions={false}
          focusInputOnSuggestionClick={true}
          getSuggestionValue={[Function]}
          highlightFirstSuggestion={false}
          id="1"
          inputProps={
            Object {
              "onChange": [Function],
              "placeholder": "What would you like to do?",
              "value": "",
            }
          }
          multiSection={false}
          onSuggestionsClearRequested={[Function]}
          onSuggestionsFetchRequested={[Function]}
          renderInputComponent={[Function]}
          renderSuggestion={[Function]}
          renderSuggestionsContainer={[Function]}
          shouldRenderSuggestions={[Function]}
          suggestions={Array []}
          theme={
            Object {
              "container": "react-autosuggest__container",
              "containerOpen": "react-autosuggest__container--open",
              "input": "react-autosuggest__input",
              "inputFocused": "react-autosuggest__input--focused",
              "inputOpen": "react-autosuggest__input--open",
              "sectionContainer": "react-autosuggest__section-container",
              "sectionContainerFirst": "react-autosuggest__section-container--first",
              "sectionTitle": "react-autosuggest__section-title",
              "suggestion": "react-autosuggest__suggestion",
              "suggestionFirst": "react-autosuggest__suggestion--first",
              "suggestionHighlighted": "react-autosuggest__suggestion--highlighted",
              "suggestionsContainer": "react-autosuggest__suggestions-container",
              "suggestionsContainerOpen": "react-autosuggest__suggestions-container--open",
              "suggestionsList": "react-autosuggest__suggestions-list",
            }
          }
        />,
        "className": "search_autosuggest",
        "role": "search",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "alwaysRenderSuggestions": false,
          "focusInputOnSuggestionClick": true,
          "getSuggestionValue": [Function],
          "highlightFirstSuggestion": false,
          "id": "1",
          "inputProps": Object {
            "onChange": [Function],
            "placeholder": "What would you like to do?",
            "value": "",
          },
          "multiSection": false,
          "onSuggestionsClearRequested": [Function],
          "onSuggestionsFetchRequested": [Function],
          "renderInputComponent": [Function],
          "renderSuggestion": [Function],
          "renderSuggestionsContainer": [Function],
          "shouldRenderSuggestions": [Function],
          "suggestions": Array [],
          "theme": Object {
            "container": "react-autosuggest__container",
            "containerOpen": "react-autosuggest__container--open",
            "input": "react-autosuggest__input",
            "inputFocused": "react-autosuggest__input--focused",
            "inputOpen": "react-autosuggest__input--open",
            "sectionContainer": "react-autosuggest__section-container",
            "sectionContainerFirst": "react-autosuggest__section-container--first",
            "sectionTitle": "react-autosuggest__section-title",
            "suggestion": "react-autosuggest__suggestion",
            "suggestionFirst": "react-autosuggest__suggestion--first",
            "suggestionHighlighted": "react-autosuggest__suggestion--highlighted",
            "suggestionsContainer": "react-autosuggest__suggestions-container",
            "suggestionsContainerOpen": "react-autosuggest__suggestions-container--open",
            "suggestionsList": "react-autosuggest__suggestions-list",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          className="search_autosuggest"
          role="search"
        >
          <Autosuggest
            alwaysRenderSuggestions={false}
            focusInputOnSuggestionClick={true}
            getSuggestionValue={[Function]}
            highlightFirstSuggestion={false}
            id="1"
            inputProps={
              Object {
                "onChange": [Function],
                "placeholder": "What would you like to do?",
                "value": "",
              }
            }
            multiSection={false}
            onSuggestionsClearRequested={[Function]}
            onSuggestionsFetchRequested={[Function]}
            renderInputComponent={[Function]}
            renderSuggestion={[Function]}
            renderSuggestionsContainer={[Function]}
            shouldRenderSuggestions={[Function]}
            suggestions={Array []}
            theme={
              Object {
                "container": "react-autosuggest__container",
                "containerOpen": "react-autosuggest__container--open",
                "input": "react-autosuggest__input",
                "inputFocused": "react-autosuggest__input--focused",
                "inputOpen": "react-autosuggest__input--open",
                "sectionContainer": "react-autosuggest__section-container",
                "sectionContainerFirst": "react-autosuggest__section-container--first",
                "sectionTitle": "react-autosuggest__section-title",
                "suggestion": "react-autosuggest__suggestion",
                "suggestionFirst": "react-autosuggest__suggestion--first",
                "suggestionHighlighted": "react-autosuggest__suggestion--highlighted",
                "suggestionsContainer": "react-autosuggest__suggestions-container",
                "suggestionsContainerOpen": "react-autosuggest__suggestions-container--open",
                "suggestionsList": "react-autosuggest__suggestions-list",
              }
            }
          />
        </form>,
        "className": "search",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Autosuggest
            alwaysRenderSuggestions={false}
            focusInputOnSuggestionClick={true}
            getSuggestionValue={[Function]}
            highlightFirstSuggestion={false}
            id="1"
            inputProps={
              Object {
                "onChange": [Function],
                "placeholder": "What would you like to do?",
                "value": "",
              }
            }
            multiSection={false}
            onSuggestionsClearRequested={[Function]}
            onSuggestionsFetchRequested={[Function]}
            renderInputComponent={[Function]}
            renderSuggestion={[Function]}
            renderSuggestionsContainer={[Function]}
            shouldRenderSuggestions={[Function]}
            suggestions={Array []}
            theme={
              Object {
                "container": "react-autosuggest__container",
                "containerOpen": "react-autosuggest__container--open",
                "input": "react-autosuggest__input",
                "inputFocused": "react-autosuggest__input--focused",
                "inputOpen": "react-autosuggest__input--open",
                "sectionContainer": "react-autosuggest__section-container",
                "sectionContainerFirst": "react-autosuggest__section-container--first",
                "sectionTitle": "react-autosuggest__section-title",
                "suggestion": "react-autosuggest__suggestion",
                "suggestionFirst": "react-autosuggest__suggestion--first",
                "suggestionHighlighted": "react-autosuggest__suggestion--highlighted",
                "suggestionsContainer": "react-autosuggest__suggestions-container",
                "suggestionsContainerOpen": "react-autosuggest__suggestions-container--open",
                "suggestionsList": "react-autosuggest__suggestions-list",
              }
            }
          />,
          "className": "search_autosuggest",
          "role": "search",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "alwaysRenderSuggestions": false,
            "focusInputOnSuggestionClick": true,
            "getSuggestionValue": [Function],
            "highlightFirstSuggestion": false,
            "id": "1",
            "inputProps": Object {
              "onChange": [Function],
              "placeholder": "What would you like to do?",
              "value": "",
            },
            "multiSection": false,
            "onSuggestionsClearRequested": [Function],
            "onSuggestionsFetchRequested": [Function],
            "renderInputComponent": [Function],
            "renderSuggestion": [Function],
            "renderSuggestionsContainer": [Function],
            "shouldRenderSuggestions": [Function],
            "suggestions": Array [],
            "theme": Object {
              "container": "react-autosuggest__container",
              "containerOpen": "react-autosuggest__container--open",
              "input": "react-autosuggest__input",
              "inputFocused": "react-autosuggest__input--focused",
              "inputOpen": "react-autosuggest__input--open",
              "sectionContainer": "react-autosuggest__section-container",
              "sectionContainerFirst": "react-autosuggest__section-container--first",
              "sectionTitle": "react-autosuggest__section-title",
              "suggestion": "react-autosuggest__suggestion",
              "suggestionFirst": "react-autosuggest__suggestion--first",
              "suggestionHighlighted": "react-autosuggest__suggestion--highlighted",
              "suggestionsContainer": "react-autosuggest__suggestions-container",
              "suggestionsContainerOpen": "react-autosuggest__suggestions-container--open",
              "suggestionsList": "react-autosuggest__suggestions-list",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
